torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2141,14 +2141,11 @@ void ProcessGroupNCCL::broadcastDumpSignal() {
   // broadcast dump signal to all other global ranks.
   broadcastSignal(globalStore_, std::string(kStoreDumpKey), globalRank());
   // signal the local rank to start dumping
-  if (shouldDump_.load()) {
-    // already signaled dump, skipping signal again and wait for the dump
-    // future.
-    return;
```

Comment:
Are we saying that the `return` here causes a second watchdog thread to early kill the process?

Reply:
yes